cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/RemoteRuntimeStoreTest.java

```diff
@@ -0,0 +1,148 @@
+ * local {@link DefaultStore} to verify the updates/writes.
+ */
+public class RemoteRuntimeStoreTest extends AppFabricTestBase {
+
+  private static DefaultStore store;
+  private static RemoteRuntimeStore runtimeStore;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    Injector injector = getInjector();
+    store = injector.getInstance(DefaultStore.class);
+    runtimeStore = injector.getInstance(RemoteRuntimeStore.class);
+  }
+
+  @Test
+  public void testSimpleCase() {
+    Id.Program flowId = Id.Flow.from(Id.Namespace.DEFAULT, "test_app", "test_flow");
+    long stopTime = System.currentTimeMillis() / 2000;
+    long startTime = stopTime - 20;
+    String pid = RunIds.generate(startTime * 1000).getId();
+    // to test null serialization (setStart can take in nullable)
+    String twillRunId = null;
+    Map<String, String> runtimeArgs = ImmutableMap.of();
+    Map<String, String> properties = ImmutableMap.of("runtimeArgs", GSON.toJson(runtimeArgs));
+    Map<String, String> systemArgs = ImmutableMap.of("a", "b");
+    RunRecordMeta initialRunRecord =
+      new RunRecordMeta(pid, startTime, null, ProgramRunStatus.RUNNING, properties, systemArgs, twillRunId);
+
+    runtimeStore.setStart(flowId, pid, startTime, twillRunId, runtimeArgs, systemArgs);
+    RunRecordMeta runMeta = store.getRun(flowId, pid);
+    Assert.assertEquals(initialRunRecord, runMeta);
+
+
+    runtimeStore.setSuspend(flowId, pid);
+    Assert.assertEquals(new RunRecordMeta(initialRunRecord, null, ProgramRunStatus.SUSPENDED),
+                        store.getRun(flowId, pid));
+
+    runtimeStore.setResume(flowId, pid);
+    Assert.assertEquals(initialRunRecord,
+                        store.getRun(flowId, pid));
+
+    // this should be a no-op, since the status is actually RUNNING, and not SUSPENDED
+    runtimeStore.compareAndSetStatus(flowId, pid, ProgramRunStatus.SUSPENDED, ProgramRunStatus.COMPLETED);
+    Assert.assertEquals(initialRunRecord,
+                        store.getRun(flowId, pid));
+
+    // this call to compareAndSetStatus will correctly mark it as COMPLETED.
+    // we don't do direct equals comparison on the run record objects, because the method internally sets the stopTime
+    runtimeStore.compareAndSetStatus(flowId, pid, ProgramRunStatus.RUNNING, ProgramRunStatus.COMPLETED);
+    RunRecordMeta runRecordMeta = store.getRun(flowId, pid);
+    Assert.assertEquals(initialRunRecord.getStartTs(), runRecordMeta.getStartTs());
+    Assert.assertEquals(initialRunRecord.getPid(), runRecordMeta.getPid());
+    Assert.assertEquals(initialRunRecord.getTwillRunId(), runRecordMeta.getTwillRunId());
+    Assert.assertEquals(ProgramRunStatus.COMPLETED, runRecordMeta.getStatus());
+  }
+
+  @Test
+  public void testWorkflowMethods() {
+    Id.Workflow workflowId = Id.Workflow.from(Id.Namespace.DEFAULT, "test_app", "test_workflow");
```

Comment:
can we use the new classes? Or is there a different PR for it? Don't want to add new deprecated code.